blog/2023-06-22-openfeature-adoption-story.md

```diff
@@ -0,0 +1,130 @@
+Open Feature lately added [Initialization](https://github.com/open-feature/spec/blob/main/specification/sections/02-providers.md#24-initialization) to the spec. This could have make it more standard implementation.
+
+## Fetching flags
+We used an option for fetching all enabled flags from the feature management service. This was not included at OpenFeature at the time, see example at [Flag Retrieval discussion](https://github.com/open-feature/ofep/issues/13#issuecomment-1337889563).
+
+## SDK used functionalities
+OpenFeature SDK has several used functionalities. What used for our case:
+* Evaluation via OpenFeature Client (getBooleanEvaluation with context and similarly for other types)
+* Initialization - see [Initialization](#initialization)
+* Events - see [Monitoring](#monitoring)
+* Fetch all flags - see [Fetching flags](#fetching-flags)
+
+###  Features Management SDK - example flow
+
+```mermaid
+sequenceDiagram
+  participant SDK as SDK (server-side)
+  participant features as Features Service
+  actor request as User Request
+  participant developer as Developer
+  participant metrics as Metrics Provider
+  note left of SDK: initialize
+  rect rgb(30,144,255)
+  note left of SDK: OpenFeature method
+  end
+  SDK->>features: fetch config https://...
+  rect rgb(127,255,0)
+  features->>SDK: Feature Flags configuration JSON
+  end
+  note left of SDK: save config in cache
+  request->>SDK: getBooleanEvaluation
+  rect rgb(30,144,255)
+  note left of SDK: OpenFeature method
+  end
+  SDK->>request: true/false from cache
+  SDK->>features: <stream> fetch config changes events https://...
+  features->>SDK: OK
+  note left of features: fetch config changes responses
+  developer->>features: Change feature flag value
+  features->>SDK: Changed event
+  SDK->>features: fetch config https://...
+  rect rgb(255,0,0)
+  features->>SDK: Error
+  end
+  rect rgb(30,144,255)
+  note left of SDK: OpenFeature Error event
+  end
+  SDK->metrics: Features Service status=Error
+  request->>SDK: getBooleanEvaluation
+  SDK->>request: true/false from cache
+```
+
+## Summary
+
+### High Level Adoption
+* Main functionallity used is for evaluating a feature flag value, which fitted our needs.
+* Adopting OpenFeature flag required adding OpenFeature SDK dependency and using it.  
+  As no other major extra steps were needed, it is done via the adoption without containers related changes.
+
+### Challenges
+Some thoughts raised during the adoption:
+* Feature Flags Configuration  
+  Considering this as the provider configuration of feature flags, e.g.  
+  ```
+  feature1=enabled
+  feature2=enabled by 50% split
+  feature3=disabled
+  feature4=enabled by customerId=4
+  ```  
+  This configuration can be represented as a scheme.  
+  I was thinking whether this configuration to be part of the standard. If so, it might be beneficial for import/export configurations accross vendors.
```

Comment:
Change Line 24 to:

Perhaps this configuration scheme could be implemented as part of the standard, adding import/export configuration across vendors?